### PR TITLE
Login Flow v2 compatibility

### PR DESCRIPTION
**Pull request by nickvergessen**
## Plan

- [x] Clean up email form
- [x] Add verification form
- [x] Clean up user form
- [x] Login Flow v1 redirect (mobile clients)
- [x] Login Flow v2 handling (desktop clients)
- [x] ~~Hide registration option in client flow when admin approval is on~~
- [x] Add tests for RegistrationController

## Screenshots

Web entering code | Web following email link
---|---
Peek 2020-07-17 11-54 | Peek 2020-07-17 11-55

Desktop client (Login Flow v2) | Android app (Login Flow v1)
---|---
Peek 2020-07-17 12-25 | Peek 2020-07-17 12-26

### CODE REVIEW

File: lib/Controller/RegisterController.php [hunk 1]
```diff
@@ -33,131 +42,218 @@ class RegisterController extends Controller {
...
+//				$this->getClientName(),
+//				IToken::PERMANENT_TOKEN
+//			);
+
+			\OC::$server->getUserSession()->login($user->getUID(), $password);
+			\OC::$server->getUserSession()->createSessionToken($this->request, $user->getUID(), $user->getUID(), $password);
+
+			/** @var LoginFlowV2Service $flowService */
+			$flowService = \OC::$server->get(LoginFlowV2Service::class);
+			$flowService->flowDone(
```
Comment (nickvergessen) [inline lib/Controller/RegisterController.php:241]: Not really possible via public api, but i zuess this is okay, as it should really be an exception, right? @rullzer

**Comment by rullzer** [comment]
I didn't check in detail but you have to hook in to both. V1 and V2.
V1 is used by mobile clients mostly.
V2 desktop + outlook etc.

**Comment by nickvergessen** [comment]
Demos for Form and Client Flow Logins

Web entering code | Web following email link
---|---
Peek 2020-07-17 11-54 | Peek 2020-07-17 11-55


Desktop client (Login Flow v2) | Android app (Login Flow v1)
---|---
Peek 2020-07-17 12-25 | Peek 2020-07-17 12-26

**Comment by nickvergessen** [comment]
> - [ ] Hide registration option in client flow when admin approval is on

This is a bit problematic it seems. Once started you reuse the session with the values. So we would never show the registration button, even if you opened it in a second window or something to just register now and set up the account later.

**Pull request merged**